ingestion/src/metadata/workflow/metadata.py

```diff
@@ -57,7 +57,15 @@ def _get_source(self) -> Source:
             )
         )
 
-        source: Source = source_class.create(self.config.source.dict(), self.metadata)
+        pipeline_name = (
+            self.ingestion_pipeline.fullyQualifiedName.__root__
+            if self.ingestion_pipeline
+            else None
+        )
+
+        source: Source = source_class.create(
+            self.config.source.dict(), self.metadata, pipeline_name
```

Comment:
I was thinking the exact same thing but I'm not a big fan of abusing the `*args` **kwargs` feature since then you lose track of things but I think we should actually have a proper `WorkflowContext` object or something similar that we use and then we can change it on one place only.

What do you think?

Reply:
generally I agree. [airflow use `**kwargs` ](https://github.com/apache/airflow/blob/main/airflow/models/baseoperator.py#L813) and IMO they found a way to manage it elegantly by raising an error for any unconsumed kwargs.

That said.. A `WorkFlowContext` will be more type-friendly and can serve as utility for dependency or configuration injection so that definitely works..